pages/api/convex-global.ts

```diff
@@ -0,0 +1,49 @@
+import { ConvexHttpClient } from "convex/browser";
+import { NextRequest as Request, NextResponse as Response } from "next/server";
+
+export const config = {
+  runtime: "edge",
+};
+
+// Make sure we use the Edge endpoints.
+let url = process.env.NEXT_PUBLIC_CONVEX_URL;
+if (!url.endsWith("edge.convex.cloud")) {
+  url = url.replace(/convex.cloud$/g, "edge.convex.cloud");
+}
+
+const convex = new ConvexHttpClient(url);
+
+const start = Date.now();
+
+export default async function api(req: Request) {
+  const count = toNumber(new URL(req.url).searchParams.get("count"));
+  const time = Date.now();
+
+  let data = null;
+  for (let i = 0; i < count; i++) {
+    const limit = 10;
+    data = await convex.query("employees")(limit);
+  }
+
+  return Response.json(
+    {
+      data,
+      queryDuration: Date.now() - time,
+      invocationIsCold: start === time,
```

Comment:
is this too aggressive/do they do the same detection elsewhere? would this return false if module loading/etc. takes more than 1ms?

Reply:
Yeah. This is what they do everywhere else :(. It can even flake if it takes less than 1ms but lets you move to the next second... I will keep it as is to be consistent.